### DISCUSSION TITLE
how to catch ctrl+enter or shift+enter shortcut in TextEdit?

### DISCUSSION THREAD

**Discussion by heng30** [slint-ui/slint#2641] [Q&A]
I try to use FocusScope inside the TextEdit. but it doesn't work. Anyone can help?

**Comment by ogoffart** [comment]
That's not easy.
I found a hack that seems to work.
But we probably need another way to query the current modifier or such things

```
import { Button, VerticalBox , LineEdit} from "std-widgets.slint";
export component Demo {
    property <bool> is_ctrl;
    width: 200px;
    height: 50px;
    FocusScope {
        VerticalLayout {
            alignment: start;
            LineEdit {
                accepted => {
                    debug(is_ctrl)
                }
            }
        }
        key-pressed(event) => {
            is_ctrl = event.modifiers.control;
            reject
        }
        key-released(event) => {
            if (event.text == Key.Control) {
                is_ctrl = false; 
            }
            reject
        }
    }
}
```

[SlintPad](https://slint-ui.com/editor/?snippet=import+%7B+Button%2C+VerticalBox+%2C+LineEdit%7D+from+%22std-widgets.slint%22%3B%0Aexport+component+Demo+%7B%0A++++property+%3Cbool%3E+is_ctrl%3B%0A++++width%3A+200px%3B%0A++++height%3A+50px%3B%0A++++FocusScope+%7B%0A++++++++VerticalLayout+%7B%0A++++++++++++alignment%3A+start%3B%0A++++++++++++LineEdit+%7B%0A++++++++++++++++accepted+%3D%3E+%7B%0A++++++++++++++++++++debug%28is_ctrl%29%0A++++++++++++++++%7D%0A++++++++++++%7D%0A++++++++%7D%0A++++++++key-pressed%28event%29+%3D%3E+%7B%0A++++++++++++is_ctrl+%3D+event.modifiers.control%3B%0A++++++++++++reject%0A++++++++%7D%0A++++++++key-released%28event%29+%3D%3E+%7B%0A++++++++++++if+%28event.text+%3D%3D+Key.Control%29+%7B%0A++++++++++++++++is_ctrl+%3D+false%3B+%0A++++++++++++%7D%0A++++++++++++reject%0A++++++++%7D%0A++++%7D%0A%7D%0A)

**Reply by heng30** [reply]
Thanks, it work for me.
